### PR TITLE
discovery/azure: Fail hard when Azure authentication parameters are missing

### PR DESCRIPTION
**Pull request by supernan1994**
fix: #4900 

Fail hard when any of 
- subscription_id
- tenant_id
- client_id
- client_secret 

is missing.

### CODE REVIEW

**Comment by supernan1994** [comment]
@simonpasquier PTAL

File: discovery/azure/azure.go [hunk 1]
```diff
@@ -80,6 +80,13 @@ type SDConfig struct {
 	RefreshInterval model.Duration     `yaml:"refresh_interval,omitempty"`
 }
 
+func (c *SDConfig) validateAuthParam(param, name string) error {
```
Comment (simonpasquier) [inline discovery/azure/azure.go]: It doesn't need to be a receiver method so you can remove `(c *SDConfig)`.

Comment (supernan1994) [reply discovery/azure/azure.go]: ok

**Comment by supernan1994** [comment]
@simonpasquier Hi😊, Should I ask someone else to review?

**Comment by simonpasquier** [comment]
@brian-brazil PTAL

**Comment by supernan1994** [comment]
@simonpasquier Thanks!

**Pull request merged**